src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/BitbucketPullRequestDecoratorTest.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    private void mockValidAnalysis() {
+        when(analysisDetails.getCommitSha()).thenReturn(COMMIT);
+        when(analysisDetails.getQualityGateStatus()).thenReturn(QualityGate.Status.OK);
+
+        Map<RuleType, Long> ruleCount = new HashMap<>();
+        ruleCount.put(RuleType.CODE_SMELL, 1L);
+        ruleCount.put(RuleType.VULNERABILITY, 2L);
+        ruleCount.put(RuleType.SECURITY_HOTSPOT, 3L);
+        ruleCount.put(RuleType.BUG, 4L);
+
+        when(analysisDetails.getRulesCount()).thenReturn(ruleCount);
+        when(analysisDetails.findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)).thenReturn(Optional.empty());
+        when(analysisDetails.findQualityGateCondition(CoreMetrics.NEW_DUPLICATED_LINES_DENSITY_KEY)).thenReturn(Optional.empty());
+        when(analysisDetails.getAnalysisDate()).thenReturn(Date.from(Instant.now()));
+        when(analysisDetails.getDashboardUrl()).thenReturn(DASHBOARD_URL);
+
+        ReportAttributes reportAttributes = mock(ReportAttributes.class);
+        when(reportAttributes.getScmPath()).thenReturn(Optional.of(ISSUE_PATH));
+
+        Component component = mock(Component.class);
+        when(component.getType()).thenReturn(Component.Type.FILE);
+        when(component.getReportAttributes()).thenReturn(reportAttributes);
+
+        DefaultIssue defaultIssue = mock(DefaultIssue.class);
+        when(defaultIssue.status()).thenReturn(Issue.STATUS_OPEN);
+        when(defaultIssue.severity()).thenReturn(Severity.CRITICAL);
+        when(defaultIssue.getLine()).thenReturn(ISSUE_LINE);
+        when(defaultIssue.key()).thenReturn(ISSUE_KEY);
+        when(defaultIssue.type()).thenReturn(RuleType.BUG);
+        when(defaultIssue.getMessage()).thenReturn(ISSUE_MESSAGE);
+        when(analysisDetails.getIssueUrl(ISSUE_KEY)).thenReturn(ISSUE_LINK);
+        when(analysisDetails.getBaseImageUrl()).thenReturn(IMAGE_URL);
+
+        PostAnalysisIssueVisitor.ComponentIssue componentIssue = mock(PostAnalysisIssueVisitor.ComponentIssue.class);
+        when(componentIssue.getIssue()).thenReturn(defaultIssue);
+        when(componentIssue.getComponent()).thenReturn(component);
+
+        PostAnalysisIssueVisitor postAnalysisIssueVisitor = mock(PostAnalysisIssueVisitor.class);
+        when(postAnalysisIssueVisitor.getIssues()).thenReturn(Collections.singletonList(componentIssue));
+
+        when(analysisDetails.getPostAnalysisIssueVisitor()).thenReturn(postAnalysisIssueVisitor);
+    }
+
+    private void verifyExpectedReport(CreateReportRequest actual) {
+        assertThat(actual.getTitle()).isEqualTo("SonarQube");
+        assertThat(actual.getResult()).isEqualTo("PASS");
+        assertThat(actual.getReporter()).isEqualTo("SonarQube");
+        assertThat(actual.getCreatedDate()).isBetween(Instant.now().minus(1, ChronoUnit.MINUTES), Instant.now());
+        assertThat(actual.getDetails()).isEqualTo("Quality Gate passed\n");
```

Comment:
Use `System.lineSeparator()` rather than `\n` to prevent this failing on Windows.